src/containers/AccessPointDetails/index.js

```diff
@@ -29,42 +52,74 @@ const AccessPointDetails = ({
   loadingFirmware,
   errorFirmware,
 }) => {
-  const [tab, setTab] = useState('general');
-
-  const tabList = [
-    {
-      key: 'general',
-      tab: 'General',
-    },
-    {
-      key: 'status',
-      tab: 'Status',
-    },
-    {
-      key: 'location',
-      tab: 'Location',
-    },
-    {
-      key: 'os',
-      tab: 'OS Stats',
-    },
-    {
-      key: 'firmware',
-      tab: 'Firmware',
-    },
-  ];
-
-  const breadCrumbs = getLocationPath(data.locationId, locations).map(location => (
-    <Breadcrumb.Item key={location.id}>{location.name}</Breadcrumb.Item>
+  const { id, tab } = useParams();
+  const history = useHistory();
+
+  const [isFormDirty, setIsFormDirty] = useState(false);
+
+  const [confirmModal, setConfirmModal] = useState(false);
+
+  const [redirectURL, setRedirectURL] = useState();
+
+  const handleOnFormChange = () => {
+    if (!isFormDirty) {
+      setIsFormDirty(true);
+    }
+  };
+
+  const handleOnBack = () => {
+    if (isFormDirty) {
+      setRedirectURL('/network/access-points');
+      setConfirmModal(true);
+    } else {
+      history.push(`/network/access-points`);
+    }
+  };
+
+  const handleTabChange = key => {
+    if (isFormDirty) {
+      setRedirectURL(`/network/access-points/${id}/${key}`);
+      setConfirmModal(true);
+    } else {
+      history.push(`/network/access-points/${id}/${key}`);
+    }
+  };
```

Comment:
these are the same functions, make them one by passing the path as a param instead